### DISCUSSION TITLE
Task farming

### DISCUSSION THREAD

**Discussion by MatthewHambley** [MetOffice/fab#304] [Discussion]
Following from a discussion about parallelism, in the original proof-of-concept we used this library: https://docs.python.org/3/library/multiprocessing.html

It supports task-farm style parallelism.

**Comment by hiker** [comment]
I just had a read - so it can fork the parent process, which should mean that the subprocess will have a copy of the parent. So that should allow a one-way flow of information, all data cached in PSyclone could be used by the child. We then only need to find a way to return new data to be cached (i.e. if a sub-process runs PSyclone, the fparser tree should be returned so it can be added to the cache
